server/console_storage.go

```diff
@@ -111,17 +123,83 @@ func (s *ConsoleServer) ListStorage(ctx context.Context, in *console.ListStorage
 		}
 		userID = &uid
 	}
-
 	var query string
+
+	collections := make([]string, 0)
+	query = "SELECT DISTINCT collection FROM storage"
+	rows, err := s.db.QueryContext(ctx, query)
+	if err != nil {
+		s.logger.Error("Error querying storage collections.", zap.Any("in", in), zap.Error(err))
+		return nil, status.Error(codes.Internal, "An error occurred while trying to list storage collections.")
+	}
+	for rows.Next() {
+		var dbCollection string
+		if err := rows.Scan(&dbCollection); err != nil {
+			_ = rows.Close()
+			s.logger.Error("Error scanning storage collections.", zap.Any("in", in), zap.Error(err))
+			return nil, status.Error(codes.Internal, "An error occurred while trying to list storage collecctions.")
+		}
+		collections = append(collections, dbCollection)
+	}
```

Comment:
I think there is a way to construct the query such that the result is only one row, where the value is an []TEXT (collections) - and you can load this directly into a string slice.

Reply:
@novabyte do you have an idea ? a quick google / stackoverflow yields only recommendation to use row scan and append

Reply:
https://stackoverflow.com/a/533326

The func `array_agg` exists in CockroachDB and Postgres 8+ so I think it's safe to use it.

Reply:
so that returns a string in the format `{coll1,coll2}` so it cannot be simply scanned into a []string but instead has to be parsed into individual names and appended to the result collection  - feels a bit fiddly for otherwise simple and robust multirow scan and append. wdyt ?